sqs/provider.go

```diff
@@ -125,6 +131,32 @@ func (s *Provider) Deprovision(ctx context.Context, deprovisionData providerifac
 }
 
 func (s *Provider) Bind(ctx context.Context, bindData provideriface.BindData) (*domain.Binding, error) {
+	bindParameters := BindParameters{}
+	if len(bindData.Details.RawParameters) > 0 {
+		decoder := json.NewDecoder(bytes.NewReader(bindData.Details.RawParameters))
+		decoder.DisallowUnknownFields()
+		if err := decoder.Decode(&bindParameters); err != nil {
+			var errStr error
+			switch castErr := err.(type) {
+				case *json.UnmarshalTypeError:
+					errStr = fmt.Errorf(
+						"Expected %#v field of bind parameters to be of type %v",
```

Comment:
question: The original error message is very similar to this, is this code an attempt to make the message more friendly to the end user? ... do we know if these messages propagate back to the user, or do they just get a "computer says no" from cloudcontroller?

Reply:
Very good question. Clearly I was hoping the former here.

Reply:
Really it's because I thought the original error exposed too many weird internal implementation details

Reply:
Yeah I do agree, it's just an annoying large bit of a code for basically amounts to removing the words "Go type" :laughing: (not your fault, Go's fault).

I would be happy simply returning a generic/static `ErrInvalidBindParams` error response to be honest (we do similar elsewhere - this makes the tests simpler too).

But I agree (if the message reaches the user) that your method of a friendly message gives a nicer UX.

Reply:
Well, it only gives a nicer UX _if_ it gets through...

Reply:
Having seen that the RDS broker just lets identical go unmarshaling errors leak through, I'm sure I can remove this.

Reply:
Except, when I just use the raw `err` response from `decoder.Decode()`... I don't seem to get a human readable string at all? Using my regular, let-the-test-fail-and-pull-the-string-to-expect-from-the-error-message technique, all I get is:

```
        Expected
            <*apiresponses.FailureResponse | 0xc0003ac900>: {                                                                                                                                                                                                         
                error: {                                                                                                                                                                                                                                              
                    Value: "array",                                                                                                                                                                                                                                   
                    Type: {size: 0x10, ptrdata: 0x8, hash: 3774831796, tflag: 7, align: 8, fieldAlign: 8, kind: 24, equal: 0x404360, gcdata: 1, str: 16322, ptrToThis: 731648},                                                                                       
                    Offset: 18,                                                                                                                                                                                                                                       
                    Struct: "UserParams",                                                                                                                                                                                                                             
                    Field: "access_policy",                                                                                                                                                                                                                           
                },                                                                                                                                                                                                                                                    
                statusCode: 400,                                                                                                                                                                                                                                      
                loggerAction: "bad-json-format",                                                                                                                                                                                                                      
                emptyResponse: false,                                                                                                                                                                                                                                 
                errorKey: "",                                                                                                                                                                                                                                         
            }                                                                                                                                                                                                                                                         
        to match error                                                                                                                                                                                                                                                
            <string>: Expected "access_policy" field of bind parameters to be of type string
```

is this just a gomega quirk or is that really a non-human-readable `err`?

Reply:
So the `error` type is an interface with an `Error() string` method, so maybe you could call `response.Error()` and then try to parse that?

Reply:
Ah! I was misinterpreting an error message when I tried to do that and getting completely the wrong end of the stick. Yup I see now.